specification/trace/semantic_conventions/cloudevents.md

```diff
@@ -0,0 +1,51 @@
+# CloudEvents
+
+This document defines how to apply semantic conventions when instrumenting code that uses [CloudEvents](https://cloudevents.io/).
```

Comment:
"uses" is to unclear here IMHO, especially for the emitter side. Should the span encompass the whole operation triggering the event (maybe using the attributes defined here just as additional attributes to the "main" span attributes, e.g. supplementing DB conventions for a SQL INSERT operation) or just the emitting itself? The latter is how it sounds currently, but I think the former would also be interesting.

Reply:
It's the latter. Since the whole messaging space is still a WIP I don't think we can even encompass a "whole operation of triggering an event", since the event can "flow" into many parts, and it is still unclear how propagation will happen in those cases.

The goal was only to offer attributes that can be added to spans that, for example, are manually created by users when they send cloudevents. Does that makes sense?

Reply:
> Since the whole messaging space is still a WIP

I thought this is no longer related to the messaging spec?

> The goal was only to offer attributes that can be added to spans that, for example, are manually created by users when they send cloudevents. Does that makes sense?

That's the case for all semantic conventions.

> I don't think we can even encompass a "whole operation of triggering an event", since the event can "flow" into many parts

It's true that for HTTP we discussed that having a separate parent per outgoing request makes sense. But still, it would be useful to somehow know which operation was the logical cause of the CloudEvent, assuming that a span for that operation exists.

Reply:
I re-worded this portion a bit in an attempt to clarify that we are only talking about sending/receiving events and not all that can come along with it. Hopefully it's clearer now. I agree with the points you made, but I feel that this whole topic is really not mature enough for us to come up with anything concrete. At least having the attributes which we know are clear can help bring interest and kick start discussions and ideas.